src/main/java/news/blog/com/service/impl/ArticleServiceImpl.java

```diff
@@ -24,16 +27,44 @@
     private final ExtendedConversionService conversionService;
 
     @Override
-    public Collection<ArticleDto> getArticles()
+    public Collection<ArticleDto> getArticles(Pageable pageable)
     {
-        return conversionService.convertMany(articleRepository.findAll(), ArticleDto.class);
+        return conversionService.convertMany(articleRepository.findAll(pageable).getContent(), ArticleDto.class);
     }
 
     @Override
-    public Collection<ArticleDto> getArticlesByUser()
+    public Collection<ArticleDto> getArticlesByUser(Pageable pageable)
     {
         UserEntity user = securityService.getUserEntity();
-        return conversionService.convertMany(articleRepository.findArticleEntitiesByUser(user), ArticleDto.class);
+        return conversionService.convertMany(articleRepository.findArticleEntitiesByUser(user, pageable).getContent(), ArticleDto.class);
+    }
+
+    @Override
+    public Collection<ArticleTagsResponseDto> getArticleTags() {
+        Collection<ArticleTagsResponseDto> articleTags = new ArrayList<>();
+        Collection<ArticleEntity> articles = articleRepository.findAll();
+        Set<String> tags = articles.stream()
+                                   .map(ArticleEntity::getTag)
+                                   .collect(Collectors.toSet());
+        tags.forEach(tag ->
+                {
+                   Long quantity = articles.stream()
```

Comment:
it's a complex operation
let's get unique tags directly from repositrory
and then we will count tags again with repository